### PR TITLE
Change BCrypt structs to use native pointers

### PR DESCRIPTION
**Pull request by AArnott**
Part of fix for #130

### CODE REVIEW

File: src/BCrypt.Tests/BCrypt.cs [hunk 1]
```diff
@@ -285,14 +285,14 @@ public unsafe void EncryptDecrypt_AesCcm()
...
                 fixed (byte* pNonce = &nonceBuffer[0])
                 {
-                    authInfo.pbNonce = new IntPtr(pNonce);
+                    authInfo.pbNonce = pNonce;
                     authInfo.cbNonce = nonceBuffer.Length;
-                    authInfo.pbTag = new IntPtr(pTagBuffer);
+                    authInfo.pbTag = pTagBuffer;
                     authInfo.cbTag = tagBuffer.Length;
 
                     // Mix up calling the IntPtr and native pointer overloads so we test both.
```
Comment (jmelosegui) [inline src/BCrypt.Tests/BCrypt.cs:293]: this is not testing the `IntPtr` any more.

**Comment by jmelosegui** [comment]
LGTM.

File: src/BCrypt.Tests/BCrypt.cs (hunk 1 above)
Comment (AArnott) [inline src/BCrypt.Tests/BCrypt.cs:293]: True. The IntPtr property now has the name `pbTag_IntPtr`. I have a couple tests sprinkled to test those. But I had to pick some change to this line (either switch the property I'm assigning to or drop the IntPtr use). In this case, I guess I picked this one. I could have gone either way. But I like my default to be what users will typically do.

**Comment by AArnott** [comment]
Thanks for the review!

**Pull request merged**